### ISSUE TITLE
Unknown string substitution fails in Run Filter

### ISSUE THREAD

**Issue by robmcmullen** [robmcmullen/peppy#785]
_[Trac time 20091015 173440Z]_ An exception is raised if an invalid `%` python string substitution is entered by the user in the Run Filter minibuffer.

**Comment by robmcmullen** [comment]
_[Trac time 20091015 173538Z]_ Entering "%S" by mistake instead of "%s" results in:

```
Traceback (most recent call last):
  File "/usr/lib64/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/_core.py", line 14614, in <lambda>
    lambda event: event.callable(*event.args, **event.kw) )
  File "/home/rob/src/peppy/peppy/actions/minibuffer.py", line 290, in performAction
    error = self.action.processMinibuffer(self, self.mode, value)
  File "/home/rob/src/peppy/peppy/jobcontrol_menu.py", line 120, in processMinibuffer
    self.mode.startCommandLine(text, expand=True)
  File "/home/rob/src/peppy/peppy/jobcontrol.py", line 255, in startCommandLine
    cmd = self.expandCommandLine(cmd)
  File "/home/rob/src/peppy/peppy/jobcontrol.py", line 247, in expandCommandLine
    cmd = cmd % filename
ValueError: unsupported format character 'S' (0x53) at index 10
```

**Comment by robmcmullen** [comment]
_[Trac time 20091020 215803Z]_ 75b07a205b6b59ee9eedb0df4cf1a4d3d9c94b3a Fixed #785: only replace %s and not all % string escapes
